### PR TITLE
Feat: Ability to limit download- and upload speed

### LINKED ISSUE #172: Option to limit Upload/Download
This would be a great feature to have!

### PR DESCRIPTION
**Pull request by OsmanAltun**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[x] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
Added possibility to limit upload or download speed

**Which issue (if any) does this pull request address?**
- Resolves #172

### CODE REVIEW

File: bin/cmd.js [hunk 1]
```diff
@@ -49,6 +49,8 @@ const options = {
     b: { alias: 'blocklist', desc: 'Load blocklist file/url', type: 'string', requiresArg: true },
     a: { alias: 'announce', desc: 'Tracker URL to announce to', type: 'string', requiresArg: true },
     q: { alias: 'quiet', desc: 'Don\'t show UI on stdout', type: 'boolean' },
+    d: { alias: 'download-limit', desc: 'Maximum download speed in kb/s', type: 'number', requiresArg: true, default: -1, defaultDescription: 'unlimited' },
+    u: { alias: 'upload-limit', desc: 'Maximum upload speed in kb/s', type: 'number', requiresArg: true, default: -1, defaultDescription: 'unlimited' },
```
Comment (alxhotel) [inline bin/cmd.js:53]:
```suggestion
    d: { alias: 'download-limit', desc: 'Maximum download speed in kB/s', type: 'number', requiresArg: true, default: -1, defaultDescription: 'unlimited' },
    u: { alias: 'upload-limit', desc: 'Maximum upload speed in kB/s', type: 'number', requiresArg: true, default: -1, defaultDescription: 'unlimited' },
```

**Review by alxhotel — CHANGES_REQUESTED** [review]

File: bin/cmd.js [hunk 2]
```diff
@@ -199,6 +201,14 @@ function init (_argv) {
     playerArgs.smplayer.push('-ontop')
   }
 
+  if (argv.downloadLimit > 0) {
+    argv.downloadLimit = argv.d = argv['download-limit'] = argv.downloadLimit * 1000
+  }
+
+  if (argv.uploadLimit > 0) {
+    argv.uploadLimit = argv.u = argv['upload-limit'] = argv.uploadLimit * 1000
```
Comment (DiegoRBaquero) [inline bin/cmd.js:209]:
```suggestion
    argv.downloadLimit = argv.d = argv['download-limit'] = argv.downloadLimit * 1024
  }

  if (argv.uploadLimit > 0) {
    argv.uploadLimit = argv.u = argv['upload-limit'] = argv.uploadLimit * 1024
```

Comment (OsmanAltun) [reply bin/cmd.js]: if you look at https://github.com/alxhotel/speed-limiter, doc says u got to do 200 * 1000 to get 200 kB/s though.

**Review by alxhotel — APPROVED** [review]
LGTM

**Comment by YOUPERT-INC** [comment]
I always appreciate every contribution you're making on this project. Could anyone let me know what's the difference between this one and the speed-limiter that @alxhotel merged a couple of weeks ago? I applied the latest Webtorrent version when speed-limiter was merged and found speed was limited as intended but it did not improve excessively high CPU usage. Does this merge have any effect in improving CPU usage?

**Comment by alxhotel** [comment]
Hi @YOUPERT-INC,

This PR adds support for the speed limiter in the CLI. So you can now limit the speed by adding flags in the `webtorrent` command.

The high CPU usage it's most likely be due to other problems, such as the one mentioned by @jimmywarting in #1558

**Comment by YOUPERT-INC** [comment]
@alxhotel Thank you very much for your kind answer! I hope I could make some donation before long because I owe so much to this project 😉

**Pull request merged**